packages/frontend/src/main/components/comments/CommentEditor.vue

```diff
@@ -84,10 +85,21 @@ export default Vue.extend({
       default: true
     }
   },
+  apollo: {
+    serverInfo: {
+      query: gql`
+        query serverInfo {
```

Comment:
This is OK for now, but it lacks types (and this is a TS component), so if you'd use `this.serverInfo` anywhere inside the component it would just have an `any` type and would need to be re-cast.

A better option is to do the following:
1. Move this query to `/packages/frontend/src/graphql/server.js` where the rest of the server queries are defined and make sure you use a globally unique query name (e.g. `query ServerFileSizeInfo { ... }`
2. Run `yarn gqlgen` to re-generate GraphQL Code Generator types
3. Import the typed query document from `'@/graphql/generated/graphql'` - you can import both the actual document to put into `query` - it'll be named ServerFileSizeInfoDocument - and also the type of the retrieved value - `ServerFileSizeInfoQuery`.
4. To type `this.serverInfo` property, create a `data()` prop like this:
```
data() {
 return {
   serverInfo: null as ServerFileSizeInfoQuery | null,
 }
}
```

(this is the suggested way to define and use queries elsewhere as well. an ever better option to use with better ergonomics would be to use the `useQuery` composable (look it up in our codebase if interested), but that's not a requirement)

Reply:
this is exactly why i wanted to discuss it with you, how to make the query nicely reusable, since its already there in 2 places. Thanks will do this.